# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/webidl/PeerConnectionImpl.webidl
# Commit: d570e7369906
# Full Hash: d570e7369906010743035437ff6ff4b4c7a34e67
# Author: Byron Campen [:bwc] <docfaraday@gmail.com>
# Date: 2022-04-06 04:35:13
# Regressor Bug: 1624989
# File Overlap Count: 1
# Description:
#   Bug 1624989: Move RTCPeerConnection's operations chain to c++ r=jib,webidl,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D141738
# ==============================================================================

diff -r 056457fa52fe -r d570e7369906 dom/webidl/PeerConnectionImpl.webidl
--- a/dom/webidl/PeerConnectionImpl.webidl	Tue Apr 05 17:41:37 2022 +0000
+++ b/dom/webidl/PeerConnectionImpl.webidl	Tue Apr 05 18:39:12 2022 +0000
@@ -15,6 +15,8 @@
 
 interface nsISupports;
 
+callback ChainedOperation = Promise<any> ();
+
 /* Must be created first. Observer events will be dispatched on the thread provided */
 [ChromeOnly,
  Exposed=Window]
@@ -44,9 +46,6 @@
   TransceiverImpl createTransceiverImpl(DOMString kind,
                                         MediaStreamTrack? track);
   [Throws]
-  boolean checkNegotiationNeeded();
-
-  [Throws]
   void replaceTrackNoRenegotiation(TransceiverImpl transceiverImpl,
                                    MediaStreamTrack? withTrack);
   [Throws]
@@ -80,6 +79,9 @@
   [Throws]
   void setConfiguration(optional RTCConfiguration config = {});
 
+  void restartIce();
+  void restartIceNoRenegotiationNeeded();
+
   /* Notify DOM window if this plugin crash is ours. */
   boolean pluginCrash(unsigned long long pluginId, DOMString name);
 
@@ -113,4 +115,7 @@
     unsigned short type, boolean ordered,
     unsigned short maxTime, unsigned short maxNum,
     boolean externalNegotiated, unsigned short stream);
+
+  Promise<any> chain(ChainedOperation op);
+  void updateNegotiationNeeded(); 
 };